File: 5.Semester/SWK5/Uebung/Beispiele Tamara Fellner/UE01_Tamara/CodeTypes/UnmanagedCodeUnmanagedTypes/Calc.cpp
#include <iostream>
#include "Calc.h"

Calc::Calc() {
	sum = 0;
	n   = 0;
}

Calc::~Calc() {
  std::cout << "~Calc" << std::endl;
}

void Calc::Add(double number) {
	sum += number;
	n++;
}

double Calc::GetSum() {
	return sum;
}

File: 5.Semester/SWK5/Uebung/Beispiele/UE01 03.10.18/Calc.cs
// bezieht sich nicht auf die Klasse sonder auf das file
using System.Reflection; // wird für das Attribut gebraucht
[assembly:AssemblyVersion("1.0.0.0")]
public class Calc
{
    protected double sum = 0;
    protected int n = 0;

    public void Add(double number)
    {
        sum += number;
        n++;
    }

    public double GetSum()
    {
        return sum;
    }
}